### PR TITLE
Replace the deprecated esprima-fb parser with flow-parser, on the RN website

### PR DESCRIPTION
**Pull request by lacker**
(I changed a ton from when I previously submitted this PR so please take another look if you already did.)

PROBLEM: the no-longer-maintained `esprima-fb` parser does not support class properties, leading our website docgen to die if we use class properties, which we're gonna do real soon now
SOLUTION: use `flow-parser` instead, which the flow team is maintaining including all the fancy-pants ES? stuff that FB uses internally.

This removes the `esprima-fb` parser from jsdocs and replaces it with `flow-parser`. It's almost the same, I checked by diffing all the parser json output and it only had a few irrelevant differences. I had to add a file of constants so that we could remove esprima-fb altogether, too.

This also adds a couple unit tests, so that we can test that jsDocs works programmatically. They don't run if you run the regular RN tests, you have to run `npm test` from the `/website/` subdirectory.

### CODE REVIEW

**Comment by facebook-github-bot-8** [comment]
@lacker updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9890/files/fb0131e8c6b49b45af0b0b4ccedfe7da510ee012..d3ad10e99c25b3b8444d99fca1a35f87752a6ab2)

File: website/server/extractDocs.js [hunk 1]
```diff
@@ -331,12 +333,25 @@ function parseAPIJsDocFormat(filepath, fileContent) {
   return json;
 }
 
+function debugDeepEqual(object1, object2) {
```
Comment (bestander) [inline website/server/extractDocs.js]: still needed?

**Comment by bestander** [comment]
How about adding to [cirlce.yml](https://github.com/facebook/react-native/blob/master/circle.yml#L54)?
```
cd website && npm test
```

**Comment by bestander** [comment]
Otherwise great job and huge impact, thanks, @lacker.

File: website/server/extractDocs.js (hunk 1 above)
Comment (lacker) [inline website/server/extractDocs.js]: Oops yeah good catch! - this is left over from the intermediate phase where I ran both doc-generators and logged all the differences. Removed it

**Comment by facebook-github-bot-6** [comment]
@lacker updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9890/files/d3ad10e99c25b3b8444d99fca1a35f87752a6ab2..13f8dfb161d3b31daa0d811ef146d6064c520db5)

**Comment by facebook-github-bot-7** [comment]
@lacker updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9890/files/13f8dfb161d3b31daa0d811ef146d6064c520db5..817e256fb430dabdffd40305e027d74db24598c1)

**Comment by bestander** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-7** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://phabricator.intern.facebook.com/D3865629) to review internal test results.

**Pull request closed without merging**